### PR TITLE
Juanro/employeedate service refactor

### PR DESCRIPTION
**Pull request by Juanro-Minne**
- added logic checks for role and model existing
- checked to stop self termination
- updated unit tests
- added fail case tests
- refactored termination service

image

### CODE REVIEW

File: HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs [hunk 1]
```diff
@@ -72,8 +103,9 @@ public async Task CheckIfExistsPassTest()
     [Fact]
     public async Task SavePassTest()
     {
+
```
Comment (madwizzard) [inline HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs]:
```suggestion
```

**Review by madwizzard — APPROVED** [review]

File: HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs [hunk 2]
```diff
@@ -8,14 +8,16 @@
 using System.Linq.Expressions;
 using RR.Tests.Data.Models;
 using Xunit;
+using HRIS.Models;
```
Comment (EduanO) [inline HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs]:
```suggestion
```

File: HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs [hunk 3]
```diff
@@ -8,14 +8,16 @@
...
 namespace HRIS.Services.Tests.Services;
 
 public class TerminationServiceUnitTests
 {
     private readonly TerminationService _terminationService;
-    private readonly Termination _terminationDto;
+    private readonly Termination _termination;
     private readonly Employee _employeeDto;
+    private readonly EmployeeDate _employeeDate;
     private readonly EmployeeType _employeeTypeDto;
```
Comment (EduanO) [inline HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs]:
```suggestion
    private readonly Employee _employee;
    private readonly EmployeeDate _employeeDate;
    private readonly EmployeeType _employeeType;
```

File: HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs [hunk 4]
```diff
@@ -28,12 +30,15 @@ public TerminationServiceUnitTests()
         _employeeTypeServiceMock = new Mock<IEmployeeTypeService>();
         _employeeServiceMock = new Mock<IEmployeeService>();
         _authServiceMock = new Mock<IAuthService>();
-        _terminationService = new TerminationService(_db.Object, _authServiceMock.Object, _employeeTypeServiceMock.Object, _employeeServiceMock.Object);
+        _employeeDate = new EmployeeDate();
```
Comment (EduanO) [inline HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs]:
```suggestion
```

File: HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs [hunk 5]
```diff
@@ -49,6 +54,32 @@ public TerminationServiceUnitTests()
         _employeeTypeDto = EmployeeTypeTestData.DeveloperType;
     }
 
+    [Fact]
+    public async Task CheckIfModelExistsReturnsTrue()
+    {
+        int testId = 1;
```
Comment (EduanO) [inline HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs]:
```suggestion
        var testId = 1;
```

File: HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs [hunk 6]
```diff
@@ -49,6 +54,32 @@ public TerminationServiceUnitTests()
...
+        var result = await _terminationService.CheckIfModelExists(testId);
+
+        Assert.True(result);
+        _db.Verify(x => x.Termination.Any(It.IsAny<Expression<Func<Termination, bool>>>()), Times.Once);
+    }
+
+    [Fact]
+    public async Task CheckIfModelExistsReturnsFalse()
+    {
+        int testId = 1;
```
Comment (EduanO) [inline HRIS.Services.Tests/Services/TerminationServiceUnitTests.cs]:
```suggestion
        var testId = 1;
```

File: HRIS.Services/Services/TerminationService.cs [hunk 7]
```diff
@@ -11,23 +12,33 @@ public class TerminationService : ITerminationService
...
+    public TerminationService(IUnitOfWork db, IEmployeeTypeService employeeTypeService, IEmployeeService employeeService, IAuthService authService, AuthorizeIdentity identity)
     {
         _db = db;
         _employeeTypeService = employeeTypeService;
         _employeeService = employeeService;
         _authService = authService;
+        _identity = identity;
+    }
+
+    public async Task<bool> CheckIfModelExists(int id)
```
Comment (EduanO) [inline HRIS.Services/Services/TerminationService.cs]:
```suggestion
    public async Task<bool> TerminationExists(int id)
```

File: HRIS.Services/Services/TerminationService.cs [hunk 8]
```diff
@@ -11,23 +12,33 @@ public class TerminationService : ITerminationService
...
+    {
+        return await _db.Termination.Any(x => x.Id == id);
     }
 
     public async Task<TerminationDto> SaveTermination(TerminationDto terminationDto)
```
Comment (EduanO) [inline HRIS.Services/Services/TerminationService.cs]:
```suggestion
    public async Task<TerminationDto> CreateTermination(TerminationDto terminationDto)
```

File: HRIS.Services/Interfaces/ITerminationService.cs [hunk 9]
```diff
@@ -24,4 +24,10 @@ public interface ITerminationService
     /// <param name="employeeId">employeeId</param>
     /// <returns>TerminationDto</returns>
     Task<TerminationDto> GetTerminationByEmployeeId(int employeeId);
-}
\ No newline at end of file
+    /// <summary>
```
Comment (EduanO) [inline HRIS.Services/Interfaces/ITerminationService.cs:27]:
```suggestion
    /// <summary>
    ///     Update an existing termination.
    /// </summary>
    /// <param name="terminationDto">The Termination DTO.</param>
    /// <returns>The updated Termination DTO.</returns>
    Task<TerminationDto> UpdateTerminationAsync(TerminationDto terminationDto);
    
    /// <summary>
```

**Review by EduanO — CHANGES_REQUESTED** [review]

**Review by MathildaShongwe15 — COMMENTED** [review]
Coverage report not attached.

**Comment by Juanro-Minne** [comment]
> Coverage report not attached.

attached

**Review by MathildaShongwe15 — APPROVED** [review]

**Review by EduanO — APPROVED** [review]

**Pull request merged**